### ISSUE TITLE
Projects are tightly coupled to tools

### ISSUE THREAD

**Issue by andylolz** [CodeClub/python-curriculum#46]
This is true for all the curricula, really – but particularly html and python.

We don’t have python installed on the machines we’re using, so I’d rather use [repl.it](http://repl.it/). However, to do that requires heavily editing the project materials just to make them make sense – since they are currently tightly coupled to IDLE.

**Comment by CodeClubRik** [comment]
Hey @andylolz 

The plan is to use [trinket.io](http://trinket.io) for HTML and Python projects. The HTML projects are being updated now, and the Python ones will be updated afterwards.

I think what will happen is that the documents will assume the use of an online tool, such as trinket, but there'll be separate guides for using 'offline' tools. Assuming the use of something like trinket actually seems to remove a lot of the tool-specific stuff from the project notes.

**Comment by martinpeck** [comment]
>> We don’t have python installed on the machines we’re using

Not sure I entirely follow. Which machines are you referring to?

Being tightly coupled to the tools is not necessarily a problem. Scratch projects will always be tied to Scratch (as an extreme example), and being tied to IDLE (or trinket, or anything else) isn't such a bad thing. It gives us a much clearer understanding of the environment that the children will perform the lesson under, and allows us to predict exactly what will happen at each stage.

Unless a particular tool is hard to provision in class rooms (IDLE might well fall into that category for schools where installing stuff isn't easy) I reckon it's OK for us to have that dependency as it allows us to write lesson plans that can better guide the children, and the volunteers.

**Comment by andylolz** [comment]
> It gives us a much clearer understanding of the environment that the children will perform the lesson under, and allows us to predict exactly what will happen at each stage.

I think you can do that without tightly coupling the project. The Scratch projects manage to do it without tightly coupling to either online or offline Scratch (we’ve used both at my club, the projects appeared to be agnostic).

> Assuming the use of something like trinket actually seems to remove a lot of the tool-specific stuff from the project notes.

Okay, well that’s great news. I (hurriedly) stripped a load of instructions out, and it worked okay: http://treadsoft.ly/CodeClubUK-Projects/en-GB/05_python/01/ASCII%20Art.html#step-1

**Comment by CodeClubRik** [comment]
I think the first project for a new language may have to include some instructions on using the tool. I did try taking the tool-specific instructions out when testing the new HTML stuff, but found that children weren't clicking through to other instructions, and therefore didn't know how to save, etc.

My compromise was to recommend online tools (i.e. trinket) and include brief instructions on loading/saving/editing/sharing etc. in the first project _only_. For those using offline tools, I've linked to another document explaining what to do.
